Source/interfaces/IDisplayInfo.h

```diff
@@ -33,43 +33,108 @@ namespace Exchange {
         };
 
         virtual uint32_t Register(INotification*) = 0;
         virtual uint32_t Unregister(INotification*) = 0;
 
-        virtual bool IsAudioPassthrough () const = 0;
-        virtual bool Connected() const = 0;
-        virtual uint32_t Width() const = 0;
-        virtual uint32_t Height() const = 0;
-        virtual uint32_t VerticalFreq() const = 0;
-        virtual HDRType Type() const = 0;
-        virtual HDCPProtectionType HDCPProtection() const = 0;
+        // @property
+        // @brief Is audioPassthrough on HDMI
+        // @param passthru: enabled/disabled
+        virtual uint32_t IsAudioPassthrough (bool& passthru /* @out */) const = 0;
+
+        // @property
+        // @brief Is HDMI connected
+        // @param isconnected: connected/disconnected
+        virtual uint32_t Connected(bool& isconnected /* @out */) const = 0;
+
+        // @property
+        // @brief Horizontal resolution of TV
+        // @param width:  width of TV in pixels
+        virtual uint32_t Width(uint32_t& width /* @out */) const = 0;
+
+        // @property
+        // @brief Vertical resolution of TV
+        // @param height:  height of TV in pixels
+        virtual uint32_t Height(uint32_t& height /* @out */) const = 0;
+
+        // @property
+        // @brief Vertical Frequency
+        // @param vf: vertical freq
+        virtual uint32_t VerticalFreq(uint32_t& vf /* @out */) const = 0;
+
+        // @property
+        // @brief TV's Extended Display Identification Data
+        // @param edid: edid byte string
+        virtual uint32_t EDID (string& edid /* @out */) const = 0;
+
+        // @property
+        // @brief HDCP protocol used for transmission
+        // @param value: protocol
+        virtual uint32_t HDCPProtection (HDCPProtectionType& value /* @out */) const = 0;
+        virtual uint32_t HDCPProtection (const HDCPProtectionType& value) = 0;
+
+        // @property
+        // @brief Video output port on the STB used for connection to TV
+        // @param name: video output port name
+        virtual uint32_t PortName (string& name /* @out */) const = 0;
+    };
+
+    /* @json */
+    struct EXTERNAL IHDRProperties : virtual public Core::IUnknown {
+        enum { ID = ID_HDR_PROPERTIES };
+
+        virtual ~IHDRProperties() { }
+
+        enum HDRType : uint8_t {
+            HDR_OFF,
+            HDR_10,
+            HDR_10PLUS,
+            HDR_DOLBYVISION,
+            HDR_TECHNICOLOR
+        };
+
+        typedef RPC::IIteratorType<HDRType, ID_HDR_ITERATOR> IHDRIterator;
+
+        // @property
+        // @brief HDR formats supported by TV
+        // @return HDRType: array of HDR formats
+        virtual uint32_t TVCapabilities(IHDRIterator& type /* out */) const;
+
+        // @property
+        // @brief HDR formats supported by STB
+        // @return HDRType: array of HDR formats
+        virtual uint32_t STBCapabilities(IHDRIterator& type /* out */) const;
```

Comment:
@pwielders : I wasn't very sure about the IIterator datatype. There were no other usages of this dataype inside Thunder. Could you please have a careful look to see if I've used it correctly?

Reply:
Hi Vinod,
This really looks good. Indeed the returning of the reference is not a good idea since this can run over process boundaries. If the otherside crashes, you could still call this but than the interface to return should be a null ptr. So I think you really should return a pointer here i.s.o. references.